tests/test_class.cpp

```diff
@@ -134,6 +134,16 @@ TEST_SUBMODULE(class_, m) {
         );
     });
 
+    struct Invalid {};
+
+    // test_type
+    m.def("check_type", [](bool valid) {
+        if (valid)
+            return py::type<DerivedClass1>();
+        else
```

Comment:
To Yannick's point, could you add a test case with a comment for type-coverted class, e.g. `int`?